packages/x-date-pickers-pro/src/DateRangeCalendar/useDragRange.ts

```diff
@@ -0,0 +1,197 @@
+    event.stopPropagation();
+    const newDate = resolveDateFromTarget(event.target, utils);
+    if (newDate) {
+      setRangeDragDay(newDate);
+      event.dataTransfer.effectAllowed = 'move';
+      setIsDragging(true);
+      const buttonDataset = (event.target as HTMLButtonElement).dataset;
+      event.dataTransfer.setData('draggingDate', buttonDataset.timestamp as string);
+      onDragStart(buttonDataset.position as DateRangePosition);
+    }
+  });
+  const handleTouchStart = useEventCallback((event: React.TouchEvent<HTMLButtonElement>) => {
+    const target = resolveElementFromTouch(event);
+    if (!target) {
+      return;
+    }
+    const newDate = resolveDateFromTarget(target, utils);
+    if (newDate) {
+      setRangeDragDay(newDate);
+      setIsDragging(true);
+      const button = event.target as HTMLButtonElement;
+      const buttonDataset = button.dataset;
+      onDragStart(buttonDataset.position as DateRangePosition);
+    }
+  });
+  const handleDragEnter = useEventCallback((event: React.DragEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    event.dataTransfer.dropEffect = 'move';
+    setRangeDragDay(resolveDateFromTarget(event.target, utils));
+  });
+  const handleTouchMove = useEventCallback((event: React.TouchEvent<HTMLButtonElement>) => {
+    const target = resolveElementFromTouch(event);
+    if (!target) {
+      return;
+    }
+    const newDate = resolveDateFromTarget(target, utils);
+    if (newDate) {
+      setRangeDragDay(newDate);
+    }
+  });
+  const handleDragLeave = useEventCallback((event: React.DragEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+  });
+  const handleDragOver = useEventCallback((event: React.DragEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    event.dataTransfer.dropEffect = 'move';
+  });
+  const handleTouchEnd = useEventCallback((event: React.TouchEvent<HTMLButtonElement>) => {
+    setRangeDragDay(null);
+    setIsDragging(false);
+    const target = resolveElementFromTouch(event, true);
+    if (!target) {
+      return;
+    }
+    const newDate = resolveDateFromTarget(target, utils);
+    if (newDate) {
+      onDrop(newDate);
+    }
+  });
+  const handleDragEnd = useEventCallback((event: React.DragEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragging(false);
+    setRangeDragDay(null);
+  });
+  const handleDrop = useEventCallback((event: React.DragEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    event.stopPropagation();
+    setIsDragging(false);
+    setRangeDragDay(null);
+    if (isSameAsDraggingDate(event)) {
+      return;
+    }
+    const newDate = resolveDateFromTarget(event.target, utils);
+    if (newDate) {
+      onDrop(newDate);
+    }
+  });
+
+  return React.useMemo(
```

Comment:
You are spreading the response from this hook in `DateRangeCalendar` so the `useMemo` has no effect.

Reply:
Changed it. Now it's no longer being spread in the wrapping hook and stops the hook from re-creating on each render. 👌